.github/workflows/macos.yml

```diff
@@ -116,27 +134,33 @@ jobs:
         uses: actions/upload-artifact@v3
         if: ${{ env.BUILD_SHARED_LIBS == 'OFF' }}
         with:
-          name: open3d-app-macosx-10_15
+          name: open3d-app-macosx-10_15-${{ runner.arch}}
           path: build/bin/open3d-app-macosx-10_15.zip
           if-no-files-found: error
 
   build-wheel:
     name: Build wheel
-    runs-on: macos-12
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       # https://github.community/t/how-to-conditionally-include-exclude-items-in-matrix-eg-based-on-branch/16853/6
       matrix:
+        # macos-12 is Intel runner, macos-14 is Apple Silicon
+        os: [macos-12, macos-14]
         python_version: ['3.8', '3.9', '3.10', '3.11']
         is_main:
           - ${{ github.ref == 'refs/heads/main' }}
         exclude:
-          - is_main: false
+          - os: macos-14
             python_version: '3.8'
-          - is_main: false
+          - os: macos-14
```

Comment:
OK to just move all macOS CI (for feature branches) to Python 3.9 for simplicity (no need to have separate options for Intel vs ARM). For main branch, we would need to keep 3.8-3.11 for Intel and 3.9-3.11 for ARM.

Reply:
The setup-python action for macOS arm64 currently doesn't work for Python 3.9 unfortunately, see [setup-python#808](https://github.com/actions/setup-python/issues/808). Pull request [python-versions#259](https://github.com/actions/python-versions/pull/259) will fix this, once it's merged we the `if` statements to exclude python < 3.10 for the arm64 runner.